build.gradle

```diff
@@ -0,0 +1,284 @@
+// define all versioned plugins here and apply in subprojects as necessary without version
+plugins {
+  id 'com.github.sherter.google-java-format' version '0.8' apply false
+  id 'net.ltgt.errorprone' version '0.6' apply false
+  id 'net.researchgate.release' version '2.7.0' apply false
+
+  // apply so we can correctly configure the test runner to be gradle at the project level
+  id 'org.jetbrains.gradle.plugin.idea-ext' version '0.7'
+}
+
+// run tests in intellij using gradle test runner
+idea.project.settings {
+  delegateActions {
+    delegateBuildRunToGradle = false
+    testRunner = 'GRADLE'
+  }
+}
+
+import net.ltgt.gradle.errorprone.CheckSeverity
+
+subprojects {
+  group 'com.google.cloud.tools'
+
+  repositories {
+    mavenCentral()
+  }
+
+  apply plugin: 'java'
+  apply plugin: 'checkstyle'
+  apply plugin: 'com.github.sherter.google-java-format'
+  apply plugin: 'net.ltgt.errorprone'
+
+  sourceCompatibility = JavaVersion.VERSION_1_8
+  targetCompatibility = JavaVersion.VERSION_1_8
+  compileJava.options.encoding = 'UTF-8'
+  compileJava.options.compilerArgs += [ '-Xlint:deprecation' ]
+  compileTestJava.options.compilerArgs += [ '-Xlint:deprecation' ]
+
+  /* PROJECT DEPENDENCY VERSIONS */
+  // define all common versioned dependencies here
+  project.ext.dependencyVersions = [
```

Comment:
I wonder if platform will help us keep this fixed between jib and jib-extensions.